Source/Rules/RuleEvaluationResult.cs

```diff
@@ -0,0 +1,64 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Dolittle. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Dolittle.Rules
+{
+    /// <summary>
+    /// Represents the result of a an evaluation of a <see cref="IRule"/>
+    /// </summary>
+    public class RuleEvaluationResult
```

Comment:
I've got a feeling we might want to generalize this.  I'd much prefer to do this than return a null.